aten/src/ATen/core/jit_type.h

```diff
@@ -1598,6 +1598,15 @@ struct CAFFE2_API ClassType : public NamedType {
   // These variants are not registered in the global class table.
   ClassTypePtr refine(at::ArrayRef<TypePtr> refined_slots) const;
 
+  TypePtr createWithContained(std::vector<TypePtr> contained_types) const override {
+    auto ptr = ClassType::create(name_, compilation_unit_);
+    AT_ASSERT(numAttributes() == contained_types.size());
+    for(size_t i = 0; i < attributeNames_.size(); ++i) {
+      ptr->addAttribute(attributeNames_[i], contained_types[i]);
```

Comment:
Could we do any validation on the types here ? One should subtype the other or vice versa right ?